src/Product/ProductRepository.php

```diff
@@ -267,9 +267,9 @@ public function find_mc_not_synced_product_ids( int $limit = -1, int $offset = 0
 			'type'       => $types,
 			'meta_query' => [
 				[
-					'key'     => ProductMetaHandler::KEY_SYNC_STATUS,
-					'compare' => '!=',
-					'value'   => SyncStatus::SYNCED,
+					'key'     => ProductMetaHandler::KEY_MC_STATUS,
+					'compare' => '=',
+					'value'   => MCStatus::NOT_SYNCED,
```

Comment:
I'm wondering if we are getting the right amount of "unsynced" products here.

Firstly it seems that `KEY_SYNC_STATUS` is no longer used after this PR, should we remove it?

Secondly based on PR #1728 it seems we are expecting to fetch "syncable but unsynced" products. In `update_product_mc_statuses` we set the MC status to `NOT_SYNCED` for all products that aren't found in MC. But we only have the meta values for all products that previously had a MC status meta key.

What happens if we have existing products, never synced, and then we install GLA. At what point do they get the initial MC status set?

Previously that was handled by triggering the job UpdateAllProducts which would walk through all the "syncable" products and then mark them as "pending" for their initial status. But if we don't trigger that job then it seems like we'd get 0 for unsynced products after initial install as they don't get a MC status set until something in a product is updated and we trigger the syncer hooks.

Reply:
Ah looking closer at the details I see `update_product_mc_statuses` works with both a combination of:
- `$all_product_ids` all products
- `$current_product_statuses` all products which have a MC status previously set

So that means the MC status will be set for all products, but the meta `ProductMetaHandler::KEY_VISIBILITY` isn't going to be set for those products. Getting syncable products is expensive that's why we have a background job to get the count. Do we need to combine that somehow instead of querying here?

Reply:
> Firstly it seems that KEY_SYNC_STATUS is no longer used after this PR, should we remove it?

I see that the key is used here: 

https://github.com/woocommerce/google-listings-and-ads/blob/56d20b558f2fc24764c22f1a5c8dfadddbdd5b19/src/Product/SyncerHooks.php#L310

So if we plan to release this earlier, before the pulling method I guess it could affect when duplicating a product.

Reply:
> So if we plan to release this earlier, before the pulling method I guess it could affect when duplicating a product.

Makes sense, I guess we should leave it to remain backwards compatible. Probably good to leave a note somewhere why we are keeping it around.

Reply:
> Do you mean fetching the count of syncable products from OptionsInterface::SYNCABLE_PRODUCTS_COUNT and then performing a simple calculation like not_synced = OptionsInterface::SYNCABLE_PRODUCTS_COUNT -synced_products? However, I think this would need updating the OptionsInterface::SYNCABLE_PRODUCTS_COUNT more frequently, as it's currently set only during user onboarding.

That's what I was thinking only on the first round, and then hoping that eventually we have meta updated so we can do a better calculation quicker. But I actually don't have a good solution there as it seems messy.

I'm currently wondering though, why do we make it difficult on ourselves to have this show "syncable but unsynced". Does the merchant even know that we are counting it like this? What do you think the effect would be if we would put just all unsynced products in there? I don't think syncable is a very clear state to the merchant as technically a product they would normally sync but is currently out of stock is temporarily classified as unsyncable until it comes back in stock. If we have it show just all unsynced products it would be much easier to explain what the count is there. At most we might need to clarify in the docs. We now have this in the docs:

> **Not synced**. These products do not appear across Google or the Google Shopping tab as free listings. They are queued for submission or may be ineligible or excluded from the product feed.

That actually sounds like it's describing "all unsynced" anyways, so might as well match that.

Reply:
> I'm currently wondering though, why do we make it difficult on ourselves to have this show "syncable but unsynced". Does the merchant even know that we are counting it like this? What do you think the effect would be if we would put just all unsynced products in there?

In my opinion, from the merchant's perspective and without checking the documentation, I would say that "Not synced" means all products that are not in the Merchant Center regardless of their visibility. 

In the follow-up issue that I am creating, I will add the item to update the documentation.